lib/java/com/google/android/material/drawable/DrawableUtils.java

```diff
@@ -190,63 +194,124 @@ private static Drawable createTintableMutatedDrawableIfNeeded(
   }
 
   /**
-   * Composites two drawables, returning a drawable instance of {@link LayerDrawable}, with the
-   * second on top of the first. If any of the drawables is null, this method will return the other.
+   * Composites two drawables, returning a drawable instance of {@link LayerDrawable},
+   * with the top layer centered.
+   * <p>
+   * If any of the drawables is null, this method will return the other.
    *
-   * @param bottomLayerDrawable the drawable to be on the first layer (bottom)
-   * @param topLayerDrawable the drawable to be on the second layer (top)
+   * @param bottomLayerDrawable the drawable to be on the bottom layer
+   * @param topLayerDrawable the drawable to be on the top layer
    */
   @Nullable
-  public static Drawable compositeTwoLayeredDrawable(
+  public static Drawable compositeDrawables(
       @Nullable Drawable bottomLayerDrawable, @Nullable Drawable topLayerDrawable) {
+
```

Comment:
Why don't you call the overloading method with desired width/height as -1? 

Also it's confusing to have overloading methods with inconsistent behavior. What's the purpose of this newly introduced method?

Reply:
Compositing with intrinsic size (without custom one) works on any API.
Compositing with custom size works only on API 23+.

The previous method was only half working, and I had to split it into two fully working methods.